github/data_source_github_enterprise_team.go

```diff
@@ -0,0 +1,135 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceGithubEnterpriseTeam() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceGithubEnterpriseTeamRead,
+
+		Schema: map[string]*schema.Schema{
+			"enterprise_slug": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The slug of the enterprise.",
+			},
+			"slug": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"team_id"},
+				Description:   "The slug of the enterprise team.",
+			},
+			"team_id": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"slug"},
+				Description:   "The numeric ID of the enterprise team.",
```

Comment:
I'm not sure how validation acts on Optional & Computed values, but if it works like I think it should, then this should remove the need to check for `team_id != 0` in the CRUD functions
```suggestion
				Description:   "The numeric ID of the enterprise team.",
				ValidateDiagFunc: validation.ToDiagFunc(validation.IntAtLeast(1))
```